src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -516,6 +516,89 @@ default Path resolve(String other) {
         return resolve(getFileSystem().getPath(other));
     }
 
+    /**
+     * Resolves a path or a sequence of paths against this path.
+     *
+     * <p> The result of this method is the same as would be obtained
+     * by resolving the first parameter path against this path, then
+     * resovling the second parameter path against the derived path,
+     * and so on until all parameters have been processed. Each single
+     * resolution in the chain behaves as specified by {@linkplain
+     * #resolve(Path) resolve}.
+     *
+     * @implSpec
+     * The default implementation is equivalent for this path to:
+     * {@snippet lang=java :
+     * Path result = resolve(first);
+     * for (Path p : more) {
+     *     result = result.resolve(p);
+     * }
+     * }
+     *
+     * @param first the first path to resolve against this path
+     *
+     * @param more additional paths to be joined to form the path
+     *
+     * @return the resulting path
+     *
+     * @throws  InvalidPathException
+     *          if the path string cannot be converted to a Path.
+     *
+     * @see #resolve(Path)
+     *
+     * @since 22
+     */
+    default Path resolve(Path first, Path... more) {
```

Comment:
It might be better to have a `(Path...)Path` overload, so that when you have an array of `Path`s or `String`s, you don’t have to manually extract the first element and copy the array using:
```java
path.resolve(paths[0], Arrays.copyOfRange(paths, 1, paths.length))
```

Reply:
Hello @ExE-Boss 

> It might be better to have a (Path...)Path

Do you mean introducing:
```
default Path resolve(Path... more) {
...
}
```
If so, then it would allow calls like:

```
Path p = ...
p.resolve();
```
(i.e. calling resolve without any parameter)
I think that would be odd and wrong for the API to allow that.

Reply:
> I think that would be odd and wrong for the API to allow that.

Agreed.